src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -35,6 +36,42 @@ using chip::bdx::TransferSession;
 
 namespace chip {
 
+// Timeout value in seconds to abort the download if there's no progress in the transfer session.
+System::Clock::Timeout mTimeout = System::Clock::kZero;
+
+static uint8_t prevPercentageComplete = 0;
+
+void StartTimeoutTimerHandler(System::Layer * systemLayer, void * appState)
```

Comment:
nit: Can we make this name more applicable to its purpose? Maybe something like OnTransferCheckTimeout or TransferCheckHandler